src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        if (this.factory == null) {
+            throw new DataProcessingException("Factory not initialized");
+        }
+        Session session = this.factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
```

Comment:
Checking if the session is null after calling `factory.openSession()` is redundant. The `openSession` method should either return a valid `Session` object or throw an exception if it fails to do so.